landingai/data_management/media.py

```diff
@@ -483,32 +484,39 @@ def _upload_folder(
     skipped_count = 0
     medias = []
     medias_with_errors = {}
+    thread_pool = ThreadPoolExecutor(max_workers=_CONCURRENCY_LIMIT)
+    tasks = []
+
     for root, _, filenames in os.walk(path):
         _LOGGER.info(f"Uploading {len(filenames)} files")
         for filename in tqdm(filenames):
             ext = os.path.splitext(filename)[-1][1:]
             if filename.lower() in _HIDDEN_FILES_TO_IGNORE:
                 pass
             elif ext.upper() in _ALLOWED_EXTENSIONS or not validate_extensions:
-                try:
-                    result = _upload_media(
-                        client,
-                        dataset_id,
-                        filename,
-                        os.path.join(root, filename),
-                        project_id,
-                        ext,
-                    )
-                    medias.append(result)
-                except DuplicateUploadError:
-                    if not tolerate_duplicate_upload:
-                        raise
-                    skipped_count += 1
-                except Exception as e:
-                    error_count += 1
-                    medias_with_errors[filename] = e
+                task = thread_pool.submit(
+                    _upload_media,
+                    client,
+                    dataset_id,
+                    filename,
+                    os.path.join(root, filename),
+                    project_id,
+                    ext,
+                )
+                tasks.append(task)
             else:
                 skipped_count += 1
+    for task in tqdm(as_completed(tasks)):
```

Comment:
For tqdm, when you start to upload a folder, it immediately shows as 100% the progress, even though we are still uploading the images

![Screenshot 2024-04-02 at 5 24 43 PM](https://github.com/landing-ai/landingai-python/assets/17306734/5aaf3eae-a12e-4905-b63c-cc3b687780c1)

Reply:
Good catch! Thanks!